File: app/src/main/java/com/lkms/data/model/java/ExperimentStep.java
package com.lkms.data.model.java;

import com.google.gson.annotations.SerializedName;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

/**
 * Đại diện cho một hàng trong bảng "ExperimentStep".
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ExperimentStep {

    @SerializedName("experimentStepId")
    private Integer experimentStepId;

    // Khóa ngoại đến Experiment
    @SerializedName("experimentId")
    private Integer experimentId;

    // Khóa ngoại đến ProtocolStep
    @SerializedName("protocolStepId")
    private Integer protocolStepId;

    @SerializedName("stepStatus")
    private String stepStatus;

    @SerializedName("stepResult")
    private String stepResult;
}

File: app/src/main/java/com/lkms/data/model/kotlin/ExperimentStep.kt
// Trong file: data/model/ExperimentStep.kt
package com.lkms.data.model.kotlin

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Đại diện cho một hàng trong bảng "ExperimentStep".
 */
@Serializable
data class ExperimentStep(
    @SerialName("experimentStepId")
    val experimentStepId: Int? = null,

    // Khóa ngoại đến Experiment
    @SerialName("experimentId")
    val experimentId: Int? = null,

    // Khóa ngoại đến ProtocolStep
    @SerialName("protocolStepId")
    val protocolStepId: Int? = null,

    @SerialName("stepStatus")
    val stepStatus: String? = null,

    @SerialName("stepResult")
    val stepResult: String? = null
)